lib/Test/Nginx/Socket.pm

```diff
@@ -2191,6 +2196,10 @@ sub gen_curl_cmd_from_req ($$) {
         my $server = $ServerAddr;
         my $port = $ServerPortForClient;
         $link = "http://$server:$port$uri";
+        if (use_http2($block) && defined $block->tls) {
+            my $server_name = $ServerName;
+            $link = "https://$server_name:$port$uri";
```

Comment:
As I mentioned in my previous round of review, the test scaffold should *automatically* generate self-signed server certificate and key files and *automatically* make use of them in the automatically generated `nginx.conf`. All such details should be transparent. Also, we can introduce a `TEST_NGINX_USE_HTTP2_TLS=1` system environment to enable this for *any* existing tests without any edits in the tests themselves?

Reply:
@agentzh I think it can be split into two parts. Listening on HTTPS/2 and making HTTPS/2 requests from Test::Nginx. 

I have tests that are using custom certificates and want to keep it that way, but would like to make HTTPS/2 request from Test::Nginx to verify them. IMO those are two different problems.

I see the workaround in ssl_certificate_by_lua tests (using cosockets), but having first class support in Test::Nginx would be nice.